.github/workflows/test.yaml

```diff
@@ -34,37 +34,49 @@ jobs:
       - name: Lint
         run: pnpm lint
 
+      - name: Validate tests
+        run: pnpm test:validate
+
       - name: Find PR Number
         uses: jwalton/gh-find-current-pr@v1
         id: findPr
 
       - name: Run tests and collect coverage
-        if: ${{ !steps.findPr.outputs.number }}
-        run: pnpm ci:test -- --json --coverage --testLocationInResults --outputFile=report.master.json
+        run: pnpm ci:test && pnpm ci:test --browser && pnpm test:merge-coverage && pnpm test:report-coverage
```

Comment:
To save CI time, we should run `--browser` and `--node` tests in parallel I CI.

Then, we'd have a third job awaiting the other two to finish and merge their coverage afterward.

I remember @Dhaiwat10 started this implementation on his first draft for this.

Reply:
Also, in @Dhaiwat10 first draft, we had 4x groups that could be run simultaneously while ensuring that all e2es (browser x node) wouldn't interfere with one another:

 - `browser:unit`
 - `browser:e2e` — requires `fuel-core` running
 - `node:unit`
 - `node:e2e` — requires `fuel-core` running

The full spec can be seen in the first draft PR:
 - https://github.com/FuelLabs/fuels-ts/pull/728